File: src/org/codepulsar/ast/statements/While.java
package org.codepulsar.ast.statements;

import org.codepulsar.ast.Expression;
import org.codepulsar.ast.Statement;

public class While extends Statement {
    private final Expression condition;
    private final Block statements;

    private final int line;

    public While(Expression condition, Block statements, int line) {
        this.condition = condition;
        this.statements = statements;
        this.line = line;
    }

    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitWhileStatement(this);
    }

    public Expression getCondition() {
        return this.condition;
    }

    public Block getStatements() {
        return this.statements;
    }

    public int getLine() {
        return this.line;
    }
}

File: src/codepulsar/ast/statement/While.cpp
#include "While.h"


Pulsar::While::While(Expression* condition, Block* statements, int line) {
    this->condition = condition;
    this->statements = statements;
    this->line = line;
}

template<typename R>
R Pulsar::While::accept(StmtVisitor<R>& visitor) {
    visitor.visitWhileStatement(this);
}

Pulsar::Expression* Pulsar::While::getCondition() {
    return this->condition;
}

Pulsar::Block* Pulsar::While::getStatements() {
    return this->statements;
}

int Pulsar::While::getLine() {
    return this->line;
}
